### PR TITLE
Alltime compare sentry error

### PR DESCRIPTION
**Pull request by EDsCODE**
## Changes

*Please describe.*  
- fixes [this](https://sentry.io/organizations/posthog/issues/2146682291/?project=1899813&query=is%3Aunassigned+is%3Aunresolved&statsPeriod=14d) sentry error
- trying to use compare on all time should be disabled. I moved this logic to the comparefilterlogic so that it's fully contained
- also fixes enum issue that was breaking `remove-shownas` flag along with [this](https://sentry.io/organizations/posthog/issues/2197605117/?project=1899813&query=is%3Aunassigned+is%3Aunresolved&statsPeriod=14d) sentry error

*If this affects the frontend, include screenshots.*  

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

### CODE REVIEW

File: frontend/src/lib/components/CompareFilter/compareFilterLogic.js [hunk 1]
```diff
@@ -26,15 +34,20 @@ export const compareFilterLogic = kea({
...
         },
     }),
     urlToAction: ({ actions }) => ({
-        '/insights': (_, { compare }) => {
-            if (compare) {
+        '/insights': (_, { compare, shown_as, date_from }) => {
+            if (compare != null) {
                 actions.setCompare(compare)
             }
+            if (shown_as === ShownAsValue.LIFECYCLE || date_from === 'all') {
```
Comment (macobo) [inline frontend/src/lib/components/CompareFilter/compareFilterLogic.js:46]: Wouldn't this make more sense as a selector rather than a separate action?

Comment (EDsCODE) [reply frontend/src/lib/components/CompareFilter/compareFilterLogic.js:46]: Hmm can you pull searchparams in selector arguments? I've never done it before but if possible yeah we could move it there.

Comment (macobo) [reply frontend/src/lib/components/CompareFilter/compareFilterLogic.js:46]: Ah. Didn't realize it was search params - do we store them somewhere else (e.g. a related logic) - if so, connecting + selector might be cleaner. 

This is nitpicking though, all else lgtm.

**Review by macobo — APPROVED** [review]
Gonna leave whether to implement selector up to you :)

File: frontend/src/lib/components/CompareFilter/compareFilterLogic.js (hunk 1 above)
Comment (EDsCODE) [reply frontend/src/lib/components/CompareFilter/compareFilterLogic.js:46]: connected to the router. I tried giving it a shot actually but it wasn't really making sense to me

**Pull request merged**